app/code/local/Excellence/Test/controllers/User/ViewController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+class Excellence_Test_User_ViewController extends Mage_Core_Controller_Front_Action
+{
+    public function historyAction()
+    { 
+        $this->loadLayout();            
+        $this->renderLayout();    
+        if( $this->getRequest()->getParam('id') > 0 ) {
+         try {
+              $model = Mage::getModel('test/test');
+
+              $model->setId($this->getRequest()->getParam('id'))->delete();
+
+             Mage::getSingleton('adminhtml/session')->addSuccess(Mage::helper('adminhtml')->__('Item was successfully deleted'));
+             $this->_redirect('*/*/history');
+            } 
+         catch (Exception $e) {
+            Mage::getSingleton('adminhtml/session')->addError($e->getMessage());
+            $this->_redirect('*/*/');
+         }
+        }
+        
+
+        /*================================Added for deleting table 2==================================*/
+        if( $this->getRequest()->getParam('id2') > 0 ) {
+         try {
+              $model = Mage::getModel('test/test2');
+
+              $model->setId($this->getRequest()->getParam('id2'))->delete();
+
+             Mage::getSingleton('adminhtml/session')->addSuccess(Mage::helper('adminhtml')->__('Item was successfully deleted'));
+             $this->_redirect('*/*/history');
+            } 
+         catch (Exception $e) {
+            Mage::getSingleton('adminhtml/session')->addError($e->getMessage());
+            $this->_redirect('*/*/');
+         }
+        }
+        /*=============================================================================================*/
+
+      }
+ public function SaveAction() 
+ {  
+   $postdata = $this->getRequest()->getPost();
+   $id = $_POST['testid'];
```

Comment:
$_POST should never be used in magento. use proper magento function for getting data from post